src/App.jsx

```diff
@@ -1,35 +1,48 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import TodoInsert from "./components/TodoInsert";
+import TodoList from "./components/TodoList";
+import TodoTemplate from "./components/TodoTemplate";
+import useTodosStore from "./stores/todoStore";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const todos = useTodosStore((state) => state.todos);
+  const setTodos = useTodosStore((state) => state.setTodos);
```

Comment:
`useTodosStore` 를 두번호출할 필요가 있을까요?

```suggestion
  const [ todos, setTodos ] = useTodosStore((state) => ([ state.todos, state.setTodos ]));
```

Reply:
배열을 사용하여 두 개의 셀렉터를 다룰 수 있군요. 생각하지 못한 방식인데 코드가 훨씬 깔끔해진 것 같습니다! 바로 반영하겠습니다.

Reply:
배열을 사용하여 셀렉터를 다루는 방식으로 리팩토링을 하니 `Maximum update depth exceeded` 문제가 발생했습니다. 해당 에러를 분석해보니 zustand는 내부적으로 `useSyncExternalStore`를 사용하며, 이 훅은 `getSnapshot` 함수가 이전과 다른 참조를 반환하면 컴포넌트를 리랜더링 한다고 합니다.
현재 셀렉터는 매 렌더링마다 `[state.todos, state.setTodos]` 라는 새로운 배열 리터럴을 생성 및 반환합니다. 내부 `state.~`의 값이 실제로 변하지 않았더라도 새로운 배열은 이전 배열과 다른 참조를 가지기에 zustand는 컴포넌트를 다시 렌더링하려고 시도합니다. 이 과정이 무한 반복되며 무한루프 문제가 발생한 것 같습니다. 셀렉터를 각각 구현하는 방식을 고수해도 되겠지만 해당 함수를 두 번 호출하는 방식은 다소 비효율적이라고 느껴집니다. 이에 대해 조금 더 탐색을 해보니 `shallow` 셀렉터라는 키워드가 있었습니다. 얕은? 비교라는 이름에서 유추 가능하듯이 각 요소에 대한 얕은 비교를 하여 무한루프를 방지하는 방법입니다. 구현은 `[배열], shallow` 로 간단히 될 것 같습니다.

Reply:
shallow 를 사용해도 여전히 에러가 발생하여 useCallback을 사용해주었습니다.
```
const [todos, setTodos] = useTodosStore(
    useCallback((state) => [state.todos, state.setTodos], []),
    shallow
  );
```
이제는 해결이 될 줄 알았는데 여전히 에러가 발생합니다.
> Warning: The result of getSnapshot should be cached to avoid an infinite loop Error Component Stack
    at App


제가 놓치고 있는 부분이 뭘까요....? 힌트를 조금만 주시면 감사하겠습니다..

Reply:
버전이 업그레이드 되면서 바뀌었나보군요. useShallow를 사용해보시죠.

https://zustand.docs.pmnd.rs/hooks/use-shallow

Reply:
해결되었습니다...감사합니다ㅜㅜ